src/emulator/storage/apis/gcloud.ts

```diff
@@ -237,11 +232,17 @@ export function createCloudEndpoints(emulator: StorageEmulator): Router {
     // Multipart upload
     let metadataRaw: string;
     let dataRaw: Buffer;
+    let name: string | undefined;
     try {
       ({ metadataRaw, dataRaw } = parseObjectUploadMultipartRequest(
         contentTypeHeader!,
         await reqBodyToBuffer(req)
       ));
+      name = getValidName(req.query, JSON.parse(metadataRaw));
+      if (name == null) {
```

Comment:
would this ever be true? name is `string | undefined`, I don't see null as an option above

Reply:
this check should catch undefined and null. I can change it to name=== undefined too, but I think this is a bit cleaner in my eyes if anyone ever introduces a null state in the future. Lmk what you think

Reply:
you still need to declare the type on L235 as nullable. In general I would prefer we avoid any type coercion if we can in typescript since it's fraught with unexpected behavior, see https://www.youtube.com/watch?v=et8xNAc2ic8

What about just using `string | null` as opposed to `string | undefined | null` if you want to keep it to a single equality check?

Reply:
yeah thats a fair point about the type coercion, I'll swap this out to name === undefined since name can only be `string | undefined` as you mentioned